### PR TITLE
Observable.flatMap;

### PR DESCRIPTION
**Pull request by johnfoconnor**

### CODE REVIEW

**Comment by bjornbytes** [comment]
Hey, thanks for the PR.  A few minor nits:

- I know I haven't documented this anywhere, but I've been wrapping comments to 100 lines instead of 80.  This should be consistent.
- This function should have an `@arg` parameter explaining that the callback should return an Observable.  This will help the documentation get correctly generated.
- If the callback isn't specified, should it have a default?  `util.identity`, perhaps?
- `flatMap` should be alphabetized before `flatten`.

This should also have tests verifying the behavior works as expected.  I'm able to do that post-merge if you'd like.

**Comment by bjornbytes** [comment]
I've made a note to create a `CONTRIBUTING.md` file to make some of this more explicit.

**Comment by johnfoconnor** [comment]
sounds good i'll address these points and get back to you. Solid work so far on this library

**Comment by johnfoconnor** [comment]
have you thought about reorganizing the "tests" folder so you can dynamically load the test files & partition/generate headers instead of hardcoding?

File: src/operators/flatMap.lua [hunk 1]
```diff
@@ -0,0 +1,10 @@
+local Observable = require 'observable'
```
Comment (bjornbytes) [inline src/operators/flatMap.lua]: `local util = require 'util'`

File: tests/flatMap.lua [hunk 2]
```diff
@@ -0,0 +1,23 @@
+describe('flatMap', function()
+  it('produces an error if its parent errors', function()
+    local observable = Rx.Observable.fromValue(''):flatMap(function(x) return x() end)
+    expect(observable.subscribe).to.fail()
+    expect(observable:flatten().subscribe).to.fail()
```
Comment (bjornbytes) [inline tests/flatMap.lua]: Should be `flatMap` instead of `flatten`.

**Comment by bjornbytes** [comment]
I'm not sure what you mean by dynamically loading the test files -- could you give an example?

Also yes it looks like `./?/init.lua` isn't in the default package path.  I'll have to modify this.

**Comment by johnfoconnor** [comment]
> I'm not sure what you mean by dynamically loading the test files -- could you give an example?

have runner.lua use `lfs` to glob the tests folder and run the tests. The output grouping can be derived from the folder structure

```
tests/
  runner.lua
  observer/
    observer.lua
  observable/
    all.lua
    amb.lua
    ...
  subscription/
      subscription.lua    
```

It's worth noting the same type of thing could be done with `concat.lua`

**Comment by bjornbytes** [comment]
Makes sense, I'll take a look.  `flatMap` looks good, thanks!

**Pull request merged**